tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -88,6 +88,28 @@ bool DuckDBPyConnection::IsJupyter() {
 	return DuckDBPyConnection::environment == PythonEnvironmentType::JUPYTER;
 }
 
+static bool IsArrowBackedDataFrame(const py::object &df) {
+	auto &import_cache = *DuckDBPyConnection::ImportCache();
+	D_ASSERT(py::isinstance(df, import_cache.pandas().DataFrame()));
+
+	py::list dtypes = df.attr("dtypes");
+	if (dtypes.empty()) {
+		return false;
+	}
+
+	auto arrow_dtype = import_cache.pandas().core.arrays.arrow.dtype.ArrowDtype();
+	for (auto &dtype : dtypes) {
+		if (!py::isinstance(dtype, arrow_dtype)) {
```

Comment:
This needs to be fixed, for now we want to treat any dataframe that contains at least a single pyarrow backed column as fully pyarrow backed, so we don't encounter any pyarrow dtypes in the pandas bind/scan code

Reply:
This did work, so I'm not sure if we just don't have tests that produce frankenstein dataframes, or we can consume them somehow.
But it's probably safer to disable this for now